### PR TITLE
pg2 chado objects

### PR DESCRIPTION
**Pull request by gildossantos**
Create new objects for chado tables for use with psycopg2.
Tweak feature object to cope with SO update.
Tweak generic psycopg2 set up, adding generic "testing" argument.
Some flake8 changes here and there.

### CODE REVIEW

File: harvdev_utils/psycopg_functions/fb_feature_classes.py [hunk 1]
```diff
@@ -587,22 +596,22 @@ def get_agr_floc(self):
...
-                    self.agr_type = 'SO:0000667'
-                    self.agr_start = self.fmin
-                    self.agr_stop = self.fmax
-                # A delin.
-                # Odd thing is that 
-                elif self.fmax - self.fmin > 1:
-                    self.agr_type = 'SO:1000032'
-                    self.agr_start = self.fmin + 1
-                    self.agr_stop = self.fmax - 1
+        # Need to reconsider portion below and split out this function into smaller less complex ones.
```
Comment (ianlongden) [inline harvdev_utils/psycopg_functions/fb_feature_classes.py:599]: Not sure why this bit is commented out. Says it needs to be split out into smaller bits, but do not see this being done? If it is now somewhere else then this needs to be deleted and not commented out. If not done yet then has this been missed?

File: harvdev_utils/psycopg_functions/fb_chado_classes.py [hunk 2]
```diff
@@ -0,0 +1,727 @@
...
+import re
+import datetime
+import strict_rfc3339
+from harvdev_utils.char_conversions import sub_sup_sgml_to_html, sgml_to_unicode
+
+log = logging.getLogger(__name__)
+
+
+# Classes
+class Author(object):
```
Comment (ianlongden) [inline harvdev_utils/psycopg_functions/fb_chado_classes.py:21]: Not sure if this is important but these are really Pub Author objects. Reading the args this is fairly obvious so maybe I am being too picky.

**Review by ianlongden — APPROVED** [review]
So generally, very good indeed, especially the docs.

Do you think this will be used in more than just your AGR dumping?
If so then it would be nice to have some tests for the classes, else I guess the running of the script will tell you if it is working or not.

Nothing serious in my comments so happy to approve.

File: harvdev_utils/psycopg_functions/fb_feature_classes.py (hunk 1 above)
Comment (gildossantos) [reply harvdev_utils/psycopg_functions/fb_feature_classes.py:599]: This bit of code was started, but not completed, and is not currently in use. Commented this out to pass flake8, but left the code in there as a reminder for when I can get back to it. I can delete it if that makes sense.

Comment (ianlongden) [reply harvdev_utils/psycopg_functions/fb_feature_classes.py:599]: Was not sure if it was something you had just forgotten or not. As this is obviously okay, happy for you to resolve it.

File: harvdev_utils/psycopg_functions/fb_chado_classes.py (hunk 2 above)
Comment (gildossantos) [reply harvdev_utils/psycopg_functions/fb_chado_classes.py]: You're right. I'll fix that.

**Comment by gildossantos** [comment]
I think this will only be used for Alliance export stuff. For anything having to do with writing to production_chado, I'm sticking with SQLAlchemy. There's a PR coming on that. If may still make sense to make tests for that stuff.

**Comment by gildossantos** [comment]
Waiting for Travis issues with pip install to be resolved before re-requesting review.

**Comment by gildossantos** [comment]
Travis is happy now - just specifying cryptography version did the trick. No more `toomanyrequests` docker errors. Requested change implemented (i.e., referring to object as Pubauthor rather than Author).

**Review by ianlongden — APPROVED** [review]

**Pull request merged**